social/pipeline/user.py

```diff
@@ -39,17 +50,11 @@ def get_username(strategy, details, user=None, *args, **kwargs):
         else:
             username = uuid4().hex
 
-        short_username = username[:max_length - uuid_length]
         final_username = slug_func(clean_func(username[:max_length]))
 
-        # Generate a unique username for current user using username
-        # as base but adding a unique hash at the end. Original
-        # username is cut to avoid any field max_length.
-        while storage.user.user_exists(username=final_username):
-            username = short_username + uuid4().hex[:uuid_length]
-            final_username = slug_func(clean_func(username[:max_length]))
```

Comment:
@jeromelefeuvre, username is usually protected with a unique constraint, this is to avoid clashing other username when registering from different sites.